app/src/main/java/com/dragic/gamehunter/viewmodel/HomeViewModel.kt

```diff
@@ -19,12 +19,35 @@ class HomeViewModel @Inject constructor(
     var dealData by mutableStateOf<List<DealViewState>>(emptyList())
         private set
 
+    private var pageNumber = 1
+
+    private var shouldIncrement = true
+
     init {
         viewModelScope.launch {
-            dealData = repository.dealData().map { it.toDealViewState() }
+            dealData = repository.dealData(pageNumber).map { it.toDealViewState() }
+        }
+    }
+
+    fun loadNextPage() {
+        viewModelScope.launch {
+            dealData = buildList {
+                addAll(dealData)
+                addAll(repository.dealData(pageNumber).map { it.toDealViewState() })
+            }
+            shouldIncrement = true
         }
     }
 
+    fun incrementPageNumber() {
+        if (shouldIncrement) {
+            pageNumber++
+            shouldIncrement = false
+        }
+    }
+
+    fun checkNotEmpty() = dealData.isNotEmpty()
```

Comment:
This is also not what I meant in the previous comment 😄 

1. Make the `dealData.isNotEmpty` check happen **only** inside the ViewModel and do not expose it to the UI. 
2. When a call to "load more" comes, deal data is empty and the first page is loading, do nothing.
3. When a call to "load more" comes, deal data is empty and the first page FAILED to load, reload it.